Source/WebCore/Modules/mediarecorder/MediaRecorderProvider.cpp

```diff
@@ -69,23 +69,46 @@ bool MediaRecorderProvider::isSupported(const String& value)
     ContentType mimeType(value);
 #if PLATFORM(COCOA)
     auto containerType = mimeType.containerType();
-    if (!equalLettersIgnoringASCIICase(containerType, "audio/mp4"_s) && !equalLettersIgnoringASCIICase(containerType, "video/mp4"_s))
-        return false;
+    if (equalLettersIgnoringASCIICase(containerType, "audio/mp4"_s) || equalLettersIgnoringASCIICase(containerType, "video/mp4"_s)) {
+        for (auto& codec : mimeType.codecs()) {
+            // FIXME: We should further validate parameters.
+            if (!startsWithLettersIgnoringASCIICase(codec, "avc1"_s) && !startsWithLettersIgnoringASCIICase(codec, "mp4a"_s))
+                return false;
+        }
+        return true;
+    }
+    return isWebMAndSupported(value);
```

Comment:
```suggestion
    return isWebMAndOpusSupported(value);
```
or just `isWebMSupported `

Reply:
`WebMSupported` would return `true` if it's MP4 and supported which isn't what we want here.

We need to know both if it's WebM and if it's supported